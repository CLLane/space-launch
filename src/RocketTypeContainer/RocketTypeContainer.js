import React from 'react';
import { Component } from 'react';
import { getRocketTypes } from '../apiCalls/apiCalls';
import { RocketCards } from '../Cards/RocketCards';

export class RocketTypeContainer extends Component {
  constructor() {
    super();
    this.state = {
      rocketType: ''
    }
  }

  async componentDidMount() {
    let rocketType = await getRocketTypes();
    let rockets = await rocketType.map(rocket => <RocketCards rocket={rocket}/>)
    this.setState({ rocketType: rockets})
  }

  render(){
    return(
      <>
        {this.state.rocketType}
      </>
    )
  }
}
