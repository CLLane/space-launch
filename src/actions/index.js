export const nextLaunchData = (launch) => ({
  type: 'NEXT_LAUNCH',
  launch,
})

export const monthLaunchData = (launches) => ({
  type: 'MONTH_LAUNCH',
  launches
})

export const rocketTypeData = (rockets) => ({
  type: 'ROCKET_TYPE',
  rockets
})

export const getMissionData = (missions) => ({
  type: 'MISSION_DATA',
  missions
})

export const getCompanyData = (companies) => ({
  type: 'COMPANY_DATA',
  companies
})