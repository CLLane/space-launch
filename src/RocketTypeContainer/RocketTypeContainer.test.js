import React from 'react';
import { shallow } from 'enzyme';
import { RocketTypeContainer, mapDispatchToProps, mapStateToProps } from "./RocketTypeContainer";
import { rocketTypeData } from "../actions/index";

describe('RocketTypeContainer', () => {
  let wrapper;
  let mockRocketTypeData = jest.fn();
  let mockRockets = [{ name: 'Falcon'}]
  let mockRocket = [{name: 'Falcon'}]

  beforeEach(() => {
    wrapper = shallow(
      <RocketTypeContainer
        rocket={mockRocket}
        rocketTypeData={mockRocketTypeData}
        rockets={mockRockets}
        key={1}
      />
    );
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an array with rocket objects', () => {
    const mockState = {
      rockets: [{name: 'falcon'}]
    }
    const expected = {
      rockets: [{ name: 'falcon' }]
    }
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch with a rocketTypeData action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = rocketTypeData(['mock'])
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.rocketTypeData(['mock'])

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})