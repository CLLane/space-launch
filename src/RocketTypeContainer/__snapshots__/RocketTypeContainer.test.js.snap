// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketTypeContainer should match a snapshot 1`] = `
<Fragment>
  <RocketCards
    rocket={
      Object {
        "name": "Falcon",
      }
    }
  />
</Fragment>
`;
